compute/eks-ec2/main.tf

```diff
@@ -2,15 +2,71 @@
 # EKS Cluster
 # --------------------------------------------------
 
+locals {
+  managed_subnets_calculated = cidrsubnets(var.eks_cluster_cidr_block, 2, 2, 2, 2)
+  cluster_reserved_cidr      = local.managed_subnets_calculated[0] # Reserved for the control plane subnets
+  managed_subnet_az_a        = local.managed_subnets_calculated[1] # Worker nodes subnet for availability zone a
+  managed_subnet_az_b        = local.managed_subnets_calculated[2] # Worker nodes subnet for availability zone b
+  managed_subnet_az_c        = local.managed_subnets_calculated[3] # Worker nodes subnet for availability zone c
+  vpc_cidr_prefix            = tonumber(substr(var.eks_cluster_cidr_block, -2, -1))
+  # This is used to determine the prefix length for the subnets, by splitting the CIDR block for the subnets
+  # into smaller chunks based on the prefix length.
+  # The calculations for VPC CIDR prefix with value 16 are ensuring backward compatibility with existing deployments.
+  # This includes removing the first element of the list of prefixes.
+  # In each subnet calculation, we ensure that at least 256 IP addresses are available for other use than prefix reservations.
+  worker_subnets_calculated = [
+    {
+      availability_zone = format("%sa", var.aws_region)
+      subnet_cidr       = local.managed_subnet_az_a
+      prefix_reservations_cidrs = local.vpc_cidr_prefix == 20 ? cidrsubnets(local.managed_subnet_az_a, 2, 2, 2) : (
+        local.vpc_cidr_prefix == 19 ? cidrsubnets(local.managed_subnet_az_a, 2, 2, 2, 3) : (
+          local.vpc_cidr_prefix == 18 ? cidrsubnets(local.managed_subnet_az_a, 2, 3, 3, 2, 3, 4) : (
+            local.vpc_cidr_prefix == 17 ? cidrsubnets(local.managed_subnet_az_a, 3, 3, 3, 3, 3, 3, 3, 4, 5) : (
```

Comment:
See previous comment:
If I have _not_ misunderstood how cidrsubnets work (🤷) then this will not work, as our starting point is already a single /24 'subnet' and we simply cannot divide it this much... 
But as it seems to work, then I must have simply misunderstood cidrsubnets